crates/mongodb-agent-common/src/query/response.rs

```diff
@@ -182,18 +207,68 @@ fn serialize_rows(
         .try_collect()
 }
 
+fn serialize_groups(
+    mode: ExtendedJsonMode,
+    path: &[&str],
+    grouping: &Grouping,
+    docs: Vec<bson::Document>,
+) -> Result<Vec<Group>> {
+    docs.into_iter()
+        .map(|doc| {
+            let dimensions_field_value = doc.get(GROUP_DIMENSIONS_KEY).ok_or_else(|| {
+                QueryResponseError::GroupMissingDimensions {
+                    path: path_to_owned(path),
+                }
+            })?;
+
+            let dimensions_array = match dimensions_field_value {
+                Bson::Array(vec) => Cow::Borrowed(vec),
+                other_bson_value => Cow::Owned(vec![other_bson_value.clone()]),
+            };
+
+            let dimensions = grouping
+                .dimensions
+                .iter()
+                .zip(dimensions_array.iter())
+                .map(|(dimension_definition, dimension_value)| {
+                    Ok(bson_to_json(
+                        mode,
+                        dimension_definition.value_type(),
+                        dimension_value.clone(),
+                    )?)
+                })
+                .collect::<Result<_>>()?;
+
+            let aggregates = serialize_aggregates(mode, path, &grouping.aggregates, doc.into())?;
+
+            // TODO: This conversion step can be removed when the aggregates map key type is
+            // changed from String to FieldName
```

Comment:
Is there a follow up ticket for this one?

Reply:
Effectively yes. That change is in the latest ndc-spec, and I have a ticket for updating the spec https://linear.app/hasura/issue/ENG-1573/[mongodb]-update-to-latest-ndc-spec-revision